src/main/java/edu/cornell/kfs/coa/dataaccess/impl/AccountDelegateGlobalLockDaoJdbc.java

```diff
@@ -0,0 +1,95 @@
+package edu.cornell.kfs.coa.dataaccess.impl;
+
+import java.util.List;
+
+import org.apache.commons.collections4.CollectionUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.springframework.jdbc.core.SingleColumnRowMapper;
+
+import edu.cornell.kfs.coa.dataaccess.AccountDelegateGlobalLockDao;
+import edu.cornell.kfs.coa.dataaccess.CuAccountDelegateGlobalDao;
+import edu.cornell.kfs.sys.util.CuSqlChunk;
+import edu.cornell.kfs.sys.util.CuSqlQuery;
+import edu.cornell.kfs.sys.util.CuSqlQueryPlatformAwareDaoBaseJdbc;
+
+public class AccountDelegateGlobalLockDaoJdbc extends CuSqlQueryPlatformAwareDaoBaseJdbc
+        implements AccountDelegateGlobalLockDao {
+
+    private static final Logger LOG = LogManager.getLogger();
+
+    private static final int IN_LIST_CHUNK_SIZE = 1000;
+
+    private CuAccountDelegateGlobalDao accountDelegateGlobalDao;
+
+    @Override
+    public String getAnyLockingDocumentNumber(List<String> lockingRepresentations, String documentNumber) {
+        LOG.debug("getAnyLockingDocumentNumber, Checking for locking document(s)");
+        if (CollectionUtils.isEmpty(lockingRepresentations)) {
+            LOG.debug("getAnyLockingDocumentNumber, No lock representations specified, skipping search");
+            return null;
+        }
+        
+        CuSqlQuery sqlQuery = createQueryForLockingDocuments(lockingRepresentations, documentNumber);
+        List<String> results = queryForValues(sqlQuery, SingleColumnRowMapper.newInstance(String.class));
+        String lockingDocumentId = CollectionUtils.isNotEmpty(results) ? results.get(0) : null;
+        
+        LOG.debug("getAnyLockingDocumentNumber, {}",
+                () -> StringUtils.isNotBlank(lockingDocumentId)
+                        ? "Found a matching lock owned by document " + lockingDocumentId
+                        : "No matching locks found");
+        
+        return lockingDocumentId;
+    }
+
+    private CuSqlQuery createQueryForLockingDocuments(List<String> lockingRepresentations, String documentNumber) {
+        return CuSqlQuery.of(
+                "SELECT DOC_HDR_ID FROM KFS.KRNS_MAINT_LOCK_T WHERE ",
+                createAndedCriteriaForDocumentIdExclusion(documentNumber),
+                "(",
+                        createOredCriteriaForLockMatching(lockingRepresentations),
+                ") AND ROWNUM <= 1"
+        );
+    }
+
+    private CuSqlChunk createAndedCriteriaForDocumentIdExclusion(String documentNumber) {
+        if (StringUtils.isNotBlank(documentNumber)) {
+            return CuSqlChunk.of(
+                    "DOC_HDR_ID <> ", CuSqlChunk.forParameter(documentNumber), " AND ");
+        } else {
+            return new CuSqlChunk();
```

Comment:
I structured the SQL generation so that if the doc-ID-limit condition ended up not being needed, the query would still have the correct format. A non-blank doc ID should result in a query like this:
```sql
SELECT DOC_HDR_ID FROM KFS.KRNS_MAINT_LOCK_T
WHERE DOC_HDR_ID <> ? AND (lock_exact_match_criteria OR lock_pattern_criteria)
AND ROWNUM <= 1;
```
A blank doc ID should result in a query like this:
```sql
SELECT DOC_HDR_ID FROM KFS.KRNS_MAINT_LOCK_T
WHERE (lock_exact_match_criteria OR lock_pattern_criteria)
AND ROWNUM <= 1;
```

Reply:
Sorry, I was trying to ask under what application conditions would we need to check for locks without the document ID

Reply:
If you're asking about the conditions where the document ID parameter might be blank, it's possible that we might not actually encounter such a situation, after further review of the calling code. However, KualiCo's lock-related DAOs contain logic to handle cases where a blank document ID was provided, so I had added that to our custom code for consistency.

If you're asking about **WHY** we need to add the not-equal-to-doc-ID check to the query in the first place, that's part of how most (if not all) of the lock checks work. When a particular document needs to check if a **DIFFERENT** document is holding one of the needed maintenance locks, it will pass its own document ID to the appropriate service/DAO method(s) so that the query ignores any locks that that document may already be holding.

Reply:
sorry for being confusing, but you answered my question in the first paragraph, we don't expect to encounter the situation but base code handles it, so we should to in this class.  that works for me, thanks.